client/blocks/reader-site-notification-settings/index.jsx

```diff
@@ -0,0 +1,176 @@
+
+		toggleSubscription( this.props.siteId );
+	};
+
+	toggleNewCommentEmail = () => {
+		const toggleSubscription = this.props.notifyOnNewComments
+			? this.props.unsubscribeToNewCommentEmail
+			: this.props.subscribeToNewCommentEmail;
+
+		toggleSubscription( this.props.siteId );
+	};
+
+	render() {
+		const { translate, notifyOnNewComments, notifyOnNewPosts } = this.props;
+
+		if ( ! this.props.siteId ) {
+			return null;
+		}
+
+		return (
+			<div className="reader-site-notification-settings">
+				<span
+					className="reader-site-notification-settings__button"
+					onClick={ this.togglePopoverVisibility }
+					ref={ this.saveSpanRef }
+				>
+					<Gridicon icon="cog" size={ 24 } ref={ this.saveIconRef } />
+					<span
+						className="reader-site-notification-settings__button-label"
+						title={ translate( 'Email settings' ) }
+					>
+						{ translate( 'Settings' ) }
+					</span>
+				</span>
+
+				<ReaderPopover
+					onClose={ this.closePopover }
+					isVisible={ this.state.showPopover }
+					context={ this.iconRef }
+					ignoreContext={ this.spanRef }
+					position={ 'bottom left' }
+					className="reader-site-notification-settings__popout"
+				>
+					<div className="reader-site-notification-settings__popout-toggle">
+						{ translate( 'New post notifications' ) }
+						<FormToggle
+							onChange={ noop }
+							checked={ false }
+							wrapperClassName="reader-site-notification-settings__popout-form-toggle"
+						/>
+						<p className="reader-site-notification-settings__popout-hint">
+							{ translate( 'Receive web and mobile notifications for new posts from this site' ) }
+						</p>
+					</div>
+
+					<div className="reader-site-notification-settings__popout-toggle">
+						{ translate( 'New post emails' ) }
+						<FormToggle onChange={ this.toggleNewPostEmail } checked={ notifyOnNewPosts } />
+					</div>
+					{ notifyOnNewPosts && (
+						<SegmentedControl>
+							<ControlItem
+								selected={ this.state.selected === 'instantly' }
+								onClick={ this.setSelected( 'instantly' ) }
+							>
+								{ translate( 'Instantly' ) }
+							</ControlItem>
+							<ControlItem
+								selected={ this.state.selected === 'daily' }
+								onClick={ this.setSelected( 'daily' ) }
+							>
+								{ translate( 'Daily' ) }
+							</ControlItem>
+							<ControlItem
+								selected={ this.state.selected === 'weekly' }
+								onClick={ this.setSelected( 'weekly' ) }
+							>
+								{ translate( 'Weekly' ) }
+							</ControlItem>
+						</SegmentedControl>
+					) }
+					<div className="reader-site-notification-settings__popout-toggle">
+						{ translate( 'New comment emails' ) }
+						<FormToggle onChange={ this.toggleNewCommentEmail } checked={ notifyOnNewComments } />
+					</div>
+				</ReaderPopover>
+			</div>
+		);
+	}
+}
+
+const mapStateToProps = ( state, ownProps ) => {
+	const follow = find( getReaderFollows( state ), { blog_ID: ownProps.siteId } );
```

Comment:
hmmm... if siteId is falsey, should we do the rest of the work?

Reply:
👍  I'll return early if siteId is falsey.